// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTCiphertext";
option optimize_for = LITE_RUNTIME;

import "Enums.proto";

message Ciphertext {
    optional uint32 version = 1;
    optional SymmetricMode mode = 2;
    optional bool text = 3;
    optional SymmetricKey key = 4;
    optional bytes iv = 5;
    optional bytes tag = 6;
    optional bytes data = 7;
}

message SymmetricKey {
    optional uint32 version = 1;
    optional SymmetricKeyType type = 2;
    optional uint32 size = 3;
    optional bytes salt = 4;
    optional uint64 operations = 5;
    optional uint64 difficulty = 6;
    optional Ciphertext key = 7;
    optional uint64 parallel = 8;
}
